### DISCUSSION TITLE
Specifying variable names in custom operator?

### DISCUSSION THREAD

**Discussion by sibeshkar** [MilesCranmer/PySR#407] [Q&A]
Hi! Thanks for your work on this -

Had a question - after training, a slice of my model's CSV looks like this : 

```
Complexity,Loss,Equation
1,0.21656719,"0.55522686"
3,0.19779082,"(0.5202965 + x3)"
4,0.18656741,"cube(0.78804094 + x3)"
5,0.17970817,"(0.44131896 + (x3 / 0.29340526))"
6,0.16866684,"cube(0.78804094 + (x3 - x2))"
7,0.16247983,"(0.4539367 + ((x3 - x2) / 0.3689541))"
8,0.15914866,"cube(0.78804094 + (x3 - (x2 - x1)))"
9,0.15100345,"(0.5554471 + (((x3 - x2) + x1) / 0.42898953))"
```

I now want to start a new run, but want to initiate it with custom intuition (via custom operators/features/constraints etc) that allows it to weigh `x3 - x2` or `0.5202965 + x3` or `lambda x : x3 - x` more during search (since these tend to appear more, and I'm determining externally which expressions appear more in the results). What's the best way to do this / is there someplace I can pass a Julia expression to make this work ? 

Happy to make any PRs/mods myself with some help, if this isn't directly possible. (I already see the issue on differential operators [here](https://github.com/MilesCranmer/SymbolicRegression.jl/issues/89), is there some native Julia way to do this today?)

**Comment by MilesCranmer** [comment]
Hi @sibeshkar,

Great question. Currently this is not possible. If you restart a search with `warm_start=True`, note that it will keep all of the expressions from the last search. Due to the crossover operation, it is more likely that, e.g., `x3 - x2` will appear in a mutated expression because it appears in your hall of fame of expressions already.

However a more general approach to this this is definitely something I would be interested in if you are up for working on it! I think e.g., https://github.com/MilesCranmer/SymbolicRegression.jl/issues/193 would be awesome to have.

Cheers,
Miles
